src/metabase/api/search.clj

```diff
@@ -106,25 +106,46 @@
        :from   (from-clause-for-model model)}
       (search.filter/build-filters model context)))
 
+(defn- collection-permission-id [model]
+  (case model
+    ("dashboard" "card" "dataset" "action") [:coalesce :trashed_from_collection_id :collection_id]
```

Comment:
Maybe this list should be defined somewhere closer to the main trash logic? (is that in `metabase.models.collection`?) Just so it's more obvious to update it in case anyone adds `trashed_from_collection_id` to another model in the future